flamedisx/likelihood.py

```diff
@@ -62,6 +60,10 @@ def __init__(
         :param n_trials:
         :param **common_param_specs:  param_name = (min, max, anchors), ...
         """
+        # Keep track of whether we traced ll after setting data
+        self.traced_after_set_data = False
+        # Only trace when we have data
+        trace_ll = False if data is None else True
```

Comment:
We can't do that since we `del data` somewhere during the init, so we'd have to go into sources and check, but then there might be different datasets for different source etc. This seems to me the cleanest solution.

Reply:
Thanks, I missed that. Agree this makes sense then. We could remove the `del data` but that might cause other problems later (as the comment notes we should only use the data in the sources after it's set, since that has the correct annotations).

Reply:
I agree that the likelihood init is getting very long, we could clean it up by initializing the sources with no data and then call `set_data` at the end (similar to the Source init), then it will deal with updating the padding, batches and also trigger the tracing.
Feel free to implement or open an issue, otherwise  let's leave it as is.